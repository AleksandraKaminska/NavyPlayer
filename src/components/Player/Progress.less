.Progress {
  display: none !important;
  @media (min-width: 768px) {
    display: flex !important;
    align-items: center;
  }
  .controller {
    cursor: pointer;
  }
}