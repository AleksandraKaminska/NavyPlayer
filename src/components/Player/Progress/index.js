import React, { Component } from 'react';
const { DZ } = window;

export default class Progress extends Component {
    constructor() {
        super()
        this.elapsed = React.createRef()
        this.duration = React.createRef()
        this.progress = React.createRef()
    }

    convertTime = time => {
        let min = Math.floor(time / 60)
        let s = Math.floor(time % 60)
        return `${min}:${(s < 10 ? '0' : '') + s}`
    }

    showPosition = () => {
        DZ.Event.subscribe('player_position', e => {
            if (this.elapsed.current && this.duration.current && this.progress.current) {
                this.elapsed.current.textContent = this.convertTime(e[0]);
                this.duration.current.textContent = this.convertTime(e[1]);
                e[1] && this.progress.current.setAttribute('value', e[0] / e[1]);
            }
        });
    }

    changeSeek = ({ target, clientX }) => {
        const { x, width } = target.getBoundingClientRect()
        DZ.player.seek((clientX - x) / width * 100);
    }

    componentDidMount() {
        this.showPosition();
    }

    render() {
        return (
            <div className="progress">
                <progress onClick={this.changeSeek} value={0} max={1} ref={this.progress} />
                <div className='time'>
                    <span className="elapsed" ref={this.elapsed}>0:00</span>
                    <span> | </span>
                    <span className="duration" ref={this.duration}>0:00</span>
                </div>
            </div>
        );
    }
}
