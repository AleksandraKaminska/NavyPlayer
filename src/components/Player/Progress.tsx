import React, { useEffect, useState } from 'react'
import { Slider, Row, Col } from 'antd'
const { DZ } = window
import './Progress.less'

export const convertTime = (time: number): string => {
  const min = Math.floor(time / 60)
  const s = Math.floor(time % 60)
  return `${min}:${(s < 10 ? '0' : '') + s}`
}

const Progress = () => {
  const [elapsed, setElapsed] = useState<number>(0)
  const [duration, setDuration] = useState<number>(0)
  const [progress, setProgress] = useState<number>(0)

  const showPosition = (): void =>
    DZ?.Event.subscribe('player_position', ([elapsed, duration]: Array<number>): void => {
      setElapsed(elapsed)
      setDuration(duration)
      if (duration) {
        const percent = (elapsed / duration) * 100
        setProgress(percent)
      }
    })

  const onChange = (value: number) => setProgress(value)

  const onAfterChange = (value: number) => DZ?.player.seek(value)

  const tipFormatter = (value?: number) => {
    if (value) {
      return convertTime((value * duration) / 100)
    }
  }

  useEffect(() => showPosition)

  return (
    <Row className="Progress">
      <Col span={2} className="elapsed">
        {convertTime(elapsed)}
      </Col>
      <Col span={20}>
        <Slider value={progress} onChange={onChange} onAfterChange={onAfterChange} tipFormatter={tipFormatter} />
      </Col>
      <Col span={1} offset={1} className="duration">
        {convertTime(duration)}
      </Col>
    </Row>
  )
}

export default Progress
