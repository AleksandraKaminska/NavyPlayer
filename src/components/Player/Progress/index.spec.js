import React from 'react'
import Progress from './index'
import renderer from 'react-test-renderer'

describe('Progress', () => {
  it('should render self', () => {
    const rendered = renderer.create(<Progress />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
