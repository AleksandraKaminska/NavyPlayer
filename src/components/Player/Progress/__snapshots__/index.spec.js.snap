// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should render self 1`] = `
<div
  className="progress"
>
  <span
    className="elapsed"
  >
    0:00
  </span>
  <progress
    max={1}
    onClick={[Function]}
    value={0}
  />
  <span
    className="duration"
  >
    0:00
  </span>
</div>
`;
